tools/syz-imagegen/imagegen.go

```diff
@@ -767,85 +767,36 @@ func runCmd(cmd string, args ...string) ([]byte, error) {
 	return osutil.RunCmd(10*time.Minute, "", cmd, args...)
 }
 
-func writeImage(image *Image, length int, segments []Segment, chdir bool) []byte {
+func writeImage(image *Image, data []byte, chdir bool) ([]byte, error) {
 	buf := new(bytes.Buffer)
 	fmt.Fprintf(buf, "# Code generated by tools/syz-imagegen. DO NOT EDIT.\n")
 	fmt.Fprintf(buf, "# requires: manual\n\n")
 	fmt.Fprintf(buf, "# %v\n\n", image)
-	fmt.Fprintf(buf, `syz_mount_image$%v(&(0x7f0000000000)='%v\x00', &(0x7f0000000100)='./file0\x00',`+
-		` 0x%x, 0x%x, &(0x7f0000000200)=[`,
-		image.fs.Name, image.fs.Name, length, len(segments))
-	addr := 0x7f0000010000
-	for i, seg := range segments {
-		if i != 0 {
-			fmt.Fprintf(buf, ", ")
-		}
-		fmt.Fprintf(buf, `{&(0x%x)="%v", 0x%x, 0x%x}`,
-			addr, hex.EncodeToString(seg.data), len(seg.data), seg.offset)
-		addr += len(seg.data)
+	compressedData, err := prog.Compress(data)
+	if err != nil {
+		return nil, err
+	}
+	b64Data, err := prog.EncodeB64(compressedData)
+	if err != nil {
+		return nil, err
 	}
+
+	addrFsOptions := 0x7f0000010000 + len(compressedData)
```

Comment:
We could just use `AUTO` for all pointers and instead of `len(compressedData)` argument.
`AUTO` allocates continuous addresses during deserialization, and assigns correct values for `len`/`const` arguments.
See sys/linux/test/* for examples. 
I guess imagegen was written before `AUTO` was introduced.